model/map/src/main/java/org/keycloak/models/map/client/MapClientProvider.java

```diff
@@ -92,18 +95,20 @@ public void updateClient() {
             /** This is runtime information and should have never been part of the adapter */
             @Override
             public Map<String, Integer> getRegisteredNodes() {
-                return clientRegisteredNodesStore.computeIfAbsent(entity.getId(), k -> new ConcurrentHashMap<>());
+                ConcurrentMap<String, Long> map = clientRegisteredNodesStore.computeIfAbsent(entity.getId(), k -> new ConcurrentHashMap<>());
```

Comment:
This could be simplified to:
```suggestion
                return clientRegisteredNodesStore.computeIfAbsent(entity.getId(), k -> new ConcurrentHashMap<>())
                        .entrySet()
                        .stream()
                        .collect(Collectors.toConcurrentMap(Map.Entry::getKey, e -> TimeAdapter.fromLongWithTimeInSecondsToIntegerWithTimeInSeconds(e.getValue())));
```

Reply:
@martin-kanis - thanks, updated it. Also wrapped the result in an unmodifiableMap.